Prepare for Renovate with reusable workflows

ghostwriter opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

ghostwriter committed:
Configure minimum supported PHP version

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable automatic-releases workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable continuous-integration workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (changes requested):
Same feedback as https://github.com/mezzio/mezzio-swoole/pull/89#pullrequestreview-888416688

In addition to that, unsure if we should target next minor, or `3.0.0` :thinking:

ghostwriter committed:
Update composer.lock

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (approved):
LGTM - what's left on the maintainer side is deciding the target branch.

weierophinney commented:
We can target 2.11.x for now, as 3.0.x is not yet finalized. I'll get that sorted shortly.

weierophinney committed:
qa: resolve Psalm errors

- Pushes test functions to a separate file and adds autoloading for it
  - Also excludes this file from Psalm analysis
- Removes unnecessary array_values call

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
